### PR TITLE
Refs #30057 -- Added more diffsettings tests

### PR DESCRIPTION
**Pull request by orlnub123**
A continuation of #10785.

I've removed the test overhaul and added the original test in a separate file and an additional test that checks if the is_overridden method shows up.

If you think the overhaul was worthwhile I can fix the style issues and add it to this PR.

### CODE REVIEW

**Comment by timgraham** [comment]
I'm okay with adding this to `test_basic`:
```
# Methods of django.conf.Settings don't appear.
self.assertNotInOutput(out, 'is_overridden = ')
```
but this and the other test are already passing without the code changes. My understanding is that you want the code changes so that lower-cased variables appear in diffsettings, but as I said in https://github.com/django/django/pull/10785#issuecomment-466101668, settings must be uppercase.

**Comment by orlnub123** [comment]
If settings are required to be uppercase shouldn't it be enforced more strictly? I propose disallowing non-uppercase setting keywords to be passed to `settings.configure()` and raising an AttributeError in `UserSettingsHolder.__getattr__()` if the name (setting) isn't uppercase.
This would also simplify the contract given to the default settings object by only ever requesting uppercase attributes (besides is_overridden).

Of course these changes would be backwards incompatible so they should start out by emitting deprecation warnings in 2.2 and eventually raise errors in 3.1 per the deprecation policy.
If it sounds reasonable I'd like to implement the above in a new ticket as it's not really related to diffsettings and focus this PR on the extra tests.

**Comment by carltongibson** [comment]
Q: Shouldn't this be a new ticket — _Cleanup `diffsettings`..._, or something? Currently we've re-opened a release blocker (which should be on 2.2) for a style change...

**Comment by orlnub123** [comment]
Sorry, I didn't realize it was a release blocker when re-opening the ticket.
If what I said above gets accepted this PR probably wouldn't even get backported to 2.2. Would it be OK to close the ticket while continuing to work on this PR?

**Comment by timgraham** [comment]
Settings must be uppercase but you can use lowercased variables to write other code in a settings file. https://github.com/django/django/blob/36300ef336e3f130a0dadc1143163ff3d23dc843/django/conf/__init__.py#L149
Disallowing non-upper values for `settings.configure()` seems reasonable. I don't think a deprecation is required.

Adding additional tests in this PR is fine.

**Comment by orlnub123** [comment]
Non-upper settings don't get added to `Settings` but `UserSettingsHolder` doesn't have any such restrictions. You could do something similar to the below and it wouldn't complain:
```py
import types
from django.conf import settings

settings.configure(types.SimpleNamespace(foo='bar'))
print(settings.foo)  # Doesn't error
```

**Comment by timgraham** [comment]
An error could be added without a deprecation since that contradicts the documentation.

**Comment by orlnub123** [comment]
Thanks for the feedback! I've made the requested changes and will open a separate ticket to deal with `settings.configure()` and `UserSettingsHolder.__getattr__()`.

**Comment by orlnub123** [comment]
The build error seems unrelated.

**Comment by timgraham** [comment]
What's the "dynamic settings" test testing that the test from 573f44d62fe1e87e2c20a74eba5e20ca9ff0ed85 doesn't cover? Does your code actually use a custom `Settings` that looks like the test?

Please remove the change in `diffsettings.py`.

**Comment by orlnub123** [comment]
The test in 573f44d uses `settings.configure()` to directly assign to the `__dict__` which has worked before 573f44d. The problem comes when you pass a different object to the default_settings argument. The reason it doesn't show up without the different object is because by default diffsettings compares the settings to global_settings, and since `settings.configure()`'s default default_settings argument _is_ global_settings, nothing odd shows up (except for default_settings itself which I've added another test for).

To sum it up the tests in 573f44d erroneously succeed without the fix, which the dynamic settings test doesn't.

**Comment by timgraham** [comment]
Thanks for the clarification.
